common/src/main/java/com/commercetools/payment/domain/PaymentServiceProvider.java

```diff
@@ -38,22 +40,29 @@
 
     /**
      * Create a function that can create a payment for the passed method Id
+     *
      * @param methodId the ID of the payment method to be used for the payment object
      * @return a function method creating the payment object for the passed method Id
+     *
+     * @throws UnsupportedOperationException if such {@code methodId} is not implemented
      */
-    Function<CreatePaymentData, CompletionStage<PaymentCreationResult>> provideCreatePaymentHandler(String methodId);
+    Function<CreatePaymentData, CompletionStage<PaymentCreationResult>> provideCreatePaymentHandler(String methodId) throws UnsupportedOperationException;
 
     /**
      * Create a function that can create a payment transaction for a payment object
      * and has the ability to handle overriding of configuration values via given key value pairs.
+     *
      * @param methodId the ID of the payment method to be used for the payment transaction object
      * @return a function method creating a payment transaction
+     *
+     * @throws UnsupportedOperationException if such {@code methodId} is not implemented
      */
-    Function<CreatePaymentTransactionData, CompletionStage<PaymentTransactionCreationResult>> provideCreatePaymentTransactionHandler(String methodId);
+    Function<CreatePaymentTransactionData, CompletionStage<PaymentTransactionCreationResult>> provideCreatePaymentTransactionHandler(String methodId) throws UnsupportedOperationException;
+
+    @Nonnull
+    default UnsupportedOperationException createUnsupportedMethodException(String methodId) throws UnsupportedOperationException {
```

Comment:
Sounds reasonable, but it was implemented like this from the beginning. We could make this method (same as `provideCreatePaymentHandler()` above) mark as `@Deprecated` and implement new ones.